pygfunction/api.py

```diff
@@ -0,0 +1,120 @@
+from pygfunction.gfunction import gFunction
+from pygfunction.boreholes import Borehole
+
+from typing import Union, List, Tuple, Dict, Iterable
+
+FloatOrList = Union[float, List[float]]
+
+
+class BoreholeFieldParameters:
+    """This class represents the borehole field and can generate inputs for pygfunction"""
+
+    def __init__(self):
+        self.height_list: List[float] = []  # TODO: Document the units
+        self.depth_list: List[float] = []
+        self.bh_radius_list: List[float] = []
+        self.tilt_angle_list: List[float] = []  # TODO: Is this degrees? And I assume from vertical?
+        self.orientation_angle_list: List[float] = []  # TODO: Same question
+        self.x_coords: List[float] = []  # TODO: What are the units here, I'm assuming meters
+        self.y_coords: List[float] = []  # TODO: Same question
+
+    def initialize_borehole_field_generic(
+            self, xy_coord_pairs: List[Tuple[float, float]],
+            height: FloatOrList, depth: FloatOrList,
+            borehole_radius: FloatOrList, tilt_angle: FloatOrList = 0,
+            orientation_angle: FloatOrList = 0):
+        """
+        Generates a set of borehole field parameters from lists of values _or_ scalar values
+        Any scalar arguments are extended to the size of the coordinates list to cover each borehole.
+        List arguments must be sized to match the size of the coordinate list.
+
+        Parameters
+        ----------
+        xy_coord_pairs
+        height
+        depth
+        borehole_radius
+        tilt_angle
+        orientation_angle
+
+        Returns
+        -------
+
+        """
+        self.x_coords = [xy_pair[0] for xy_pair in xy_coord_pairs]
+        self.y_coords = [xy_pair[1] for xy_pair in xy_coord_pairs]
+        num_bh = len(xy_coord_pairs)
+        self.height_list = self.setup_list(height, num_bh)
+        self.depth_list = self.setup_list(depth, num_bh)
+        self.bh_radius_list = self.setup_list(borehole_radius, num_bh)
+        self.tilt_angle_list = self.setup_list(tilt_angle, num_bh)
+        self.orientation_angle_list = self.setup_list(orientation_angle, num_bh)
```

Comment:
Calling this flexible initialize method will validate the data and store it in the right lists internally, ready to be exported to pygfunction.  I would imagine a method later called "add_borehole" that could then be called after this to add a single borehole to the list and get a modified g-function or something.